drive/drive.go

```diff
@@ -391,6 +403,17 @@ func (d *Client) Rename(object *APIObject, OldParent string, NewParent string, N
 func (d *Client) mapFileToObject(file *gdrive.File) (*APIObject, error) {
 	Log.Tracef("Converting Google Drive file: %v", file)
 
+	actualId := file.Id
+	if file.MimeType == shortcutMimeType && file.ShortcutDetails != nil {
+		actualFile, err := d.GetFileById(file.ShortcutDetails.TargetId)
+		if err != nil {
+			return nil, err
+		}
+		actualId = actualFile.Id
+		file.MimeType = actualFile.MimeType
+		file.Size = actualFile.Size
```

Comment:
This changes the input param file which is bad. Instead we could use a copy of the file instead:

```diff
diff --git a/drive/drive.go b/drive/drive.go
index 5af076a..fadfce3 100644
--- a/drive/drive.go
+++ b/drive/drive.go
@@ -403,18 +403,16 @@ func (d *Client) Rename(object *APIObject, OldParent string, NewParent string, N
 func (d *Client) mapFileToObject(file *gdrive.File) (*APIObject, error) {
        Log.Tracef("Converting Google Drive file: %v", file)

-       actualId := file.Id
+       var err error
+       targetFile := file
        if file.MimeType == shortcutMimeType && file.ShortcutDetails != nil {
-               actualFile, err := d.GetFileById(file.ShortcutDetails.TargetId)
+               targetFile, err = d.GetFileById(file.ShortcutDetails.TargetId)
                if err != nil {
                        return nil, err
                }
-               actualId = actualFile.Id
-               file.MimeType = actualFile.MimeType
-               file.Size = actualFile.Size
        }

-       lastModified, err := time.Parse(time.RFC3339, file.ModifiedTime)
+       lastModified, err := time.Parse(time.RFC3339, targetFile.ModifiedTime)
        if nil != err {
                Log.Debugf("%v", err)
                Log.Warningf("Could not parse last modified date for object %v (%v)", file.Id, file.Name)
@@ -429,11 +427,11 @@ func (d *Client) mapFileToObject(file *gdrive.File) (*APIObject, error) {
        return &APIObject{
                ObjectID:     file.Id,
                Name:         file.Name,
-               IsDir:        file.MimeType == folderMimeType,
+               IsDir:        targetFile.MimeType == folderMimeType,
                LastModified: lastModified,
-               Size:         uint64(file.Size),
-               DownloadURL:  fmt.Sprintf("https://www.googleapis.com/drive/v3/files/%v?alt=media", actualId),
+               Size:         uint64(targetFile.Size),
+               DownloadURL:  fmt.Sprintf("https://www.googleapis.com/drive/v3/files/%v?alt=media", targetFile.Id),
                Parents:      parents,
                CanTrash:     file.Capabilities.CanTrash,
-       }, nil
+       }, err
 }
```

This also uses `lastModifiedTime` from the `targetFile` instead of the shortcut, which I think makes more sense.

@RoyXiang What do you think?

Reply:
You're right, the input param `file` should not be modified.

But for the `lastModifiedTime`, since the content of a media file would rarely be modified, a renaming of the filename would be a more-often modification, I would like to take the one of the shortcut.

What do you think?

Reply:
@RoyXiang Normally, the LastModified time of a file doesn't change, if you rename it, only if the content is changed. So I think basing it on the target file makes more sense.

Reply:
@felixbuenemann I see. The requested changes have been made.